File: 6_kyu/Begin_your_day_with_a_challenge_but_an_easy_one.c
#include <stddef.h>
#include <stdio.h>
#include <string.h>

// Code puzzle.
void one_two_three(const unsigned char n, char a[], char b[]) {
  // generate string a
  const int nines = n / 9;
  const int remain = n % 9;
  memset(a, '9', nines);
  a[nines] = 0;
  if( nines == 0 || remain > 0 )
    a[nines] = '0' + remain;
  a[nines+1] = 0;
  // generate string b
  if( n == 0 )
  {
    strcpy(b, "0");
  }
  else
  {
    memset(b, '1', n);
    b[n] = 0;
  }
}

File: 6_kyu/Begin_your_day_with_a_challenge_but_an_easy_one.py
def one_two_three(n: int) -> list[int,int]:
    a = 0
    for _ in range(n//9): a = 10*a + 9
    if n % 9 > 0:  a = 10*a + n % 9
    b = 0
    for _ in range(n): b = 10*b + 1
    return [a, b]
    